net/HttpRequest.hpp

```diff
@@ -2181,6 +2181,7 @@ class ServerSession final : public ProtocolHandlerInterface
 
                     close(_fd);
                     _fd = -1;
+                    socket->flush();
```

Comment:
Should we do this before closing `_fd` ?

Reply:
There is no really 'synchronous' flush method surely, and there should not be. We need a clean asynchronous close mechanism. I worry that people think methods are synchronous - prolly we should rename 'flush' to 'sendWhatWeCan' or something:

commit 40b1aa107792983c6018d0940d728f6733b2a3e3
Author: Ashod Nakashian <ashod.nakashian@collabora.co.uk>
Date:   Sat Mar 20 16:29:16 2021 -0400

    wsd: add member to Socket to flush safely

This looks broken to me ...

Reply:
On the specific use of flush in this pr: https://github.com/CollaboraOnline/online/pull/11982 for that